shared/src/main/scala/mlscript/syntax.scala

```diff
@@ -96,8 +108,13 @@ sealed abstract class NullaryType                        extends Type
 
 case object Top                                          extends NullaryType
 case object Bot                                          extends NullaryType
+
+// Literal type or singleton type i.e.
+// `type t: 0` is a type with only one possible value
```

Comment:
The syntax `type t: 0` means nothing.
```suggestion
// type `0` is a type with only one possible value
```